### PR TITLE
Write lock for savemat

### LINKED ISSUE #120: Errors when running many Oct2Py objects concurrently
When I try to run multiple Oct2Py instances concurrently, either with joblib / multi-threading or Apache Beam, I've been getting errors from Oct2Py that only occur when I make a large total number of Octave function calls (more than several hundred).

The errors look like:
```
Oct2PyError: Oct2Py tried to run:
"""
load /tmp/tmpMat3JD/writer.mat "A__" "B__" "C__" "D__" "E__" "F__" "G__" "H__" "I__" "J__" "K__" "L__" "M__"
"""
Octave returned:
error: load: invalid element type = 0
```

with tracebacks of the form:
```
  File ".../oct2py/dynamic.py", line 32, in __call__
    return self._ref()._call(self.name, *inputs, **kwargs)
  File ".../oct2py/core.py", line 460, in _call
    data = self.eval(cmd, temp_dir=temp_dir, **eval_kwargs)
  File ".../oct2py/core.py", line 305, in eval
    out_file=self._reader.out_file)
  File ".../oct2py/core.py", line 626, in evaluate
    raise Oct2PyError(msg)
```

Any idea what's going on here? ~~My best guess is that Python may be running out of temp directories and overwriting old files???~~ This feels like a race condition, but I can't find any non-threadsafe operations (tempfile.mkdtemp *is* allegedly threadsafe).

### PR DESCRIPTION
**Pull request by shoyer**
Fixes #120

### CODE REVIEW

File: oct2py/io.py [hunk 1]
```diff
@@ -17,6 +18,9 @@
 from .utils import Oct2PyError
 
 
+_WRITE_LOCK = threading.lock()
```
Comment (blink1073) [inline oct2py/io.py]: I think this needs to be `threading.Lock()`:  https://docs.python.org/2/library/threading.html#threading.Lock

Comment (shoyer) [inline oct2py/io.py]: Oops, that's what I get for not running unit tests before sending a PR :)

**Comment by blink1073** [comment]
Thanks!

**Comment by blink1073** [comment]
Released in 4.0.6.

**Comment by shoyer** [comment]
Thanks @blink1073!

**Pull request merged**